File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/IcePlateauTeleportSpell.java
package plugin.interaction.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/28/2017
 */
public class IcePlateauTeleportSpell implements TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 89;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 3, LAW_RUNE, 3, WATER_RUNE, 8);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2972, 3873, 0);
	}
	
	@Override
	public int spellId() {
		return 51;
	}
	
	@Override
	public double exp() {
		return 96;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/IcePlateauTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/28/2017
 */
class IcePlateauTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 89
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 3, MagicConstants.LAW_RUNE, 3, MagicConstants.WATER_RUNE, 8)
    }

    override fun destination(): WorldTile {
        return WorldTile(2972, 3873, 0)
    }

    override fun spellId(): Int {
        return 51
    }

    override fun exp(): Double {
        return 96.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}